Add docker private registry support

secat opened the pull request:
This PR addresses #43 and add support for a Docker Private Registry deployed on premise.

secat commented:
I am currently working on fixing the tests.

stevesloka commented:
@secat I'll check this out soon, just had a baby last week so a bit behind. Thanks!

secat commented:
@stevesloka No problem, I was exactly in the same situation last year! I fully understand your situation!

stevesloka commented:
Hey @secat would you mind rebasing off of master? There are some conflicts now, then I'll do a quick test and get this out. Also I can add support in minikube's addon UI.

secat commented:
Ok, I will work on it today.

secat commented:
@stevesloka Friendly reminder: this PR is ready for review and has been rebase with the latest changes.

stevesloka commented:
Why did the makefile change so much? When I try and just build I get the following: 

```
 steve@phippy  ~/godev/src/github.com/upmc-enterprises/registry-creds   add-docker-private-registry-support  make build
Building: bin/registry-creds
GOARCH=amd64 CGO_ENABLED=0 go install
go install os/user: open /usr/local/go/pkg/linux_amd64/os/user.a: permission denied
go install net: open /usr/local/go/pkg/linux_amd64/net.a: permission denied
Makefile:37: recipe for target 'bin/registry-creds' failed
make: *** [bin/registry-creds] Error 1
```

stevesloka commented:
I get the same on my mac:

```
~/g/s/g/u/registry-creds ❯❯❯ make build                                                                       add-docker-private-registry-support ◼
Building: bin/registry-creds
GOARCH=amd64 CGO_ENABLED=0 go install
go install os/user: open /usr/local/go/pkg/darwin_amd64/os/user.a: permission denied
go install net: open /usr/local/go/pkg/darwin_amd64/net.a: permission denied
make: *** [bin/registry-creds] Error 1
```

stevesloka commented on Makefile:
```diff
@@ -2,24 +2,66 @@
+GOFLAGS := -a -installsuffix cgo
+
+# docker build arguments for internal proxy
+ifneq ($(http_proxy),)
```
Is this to push to a local registry? I don't see why that's needed.

secat replied:
It is to build the `dockerfile`: it will not build if working behind a corporate proxy.

stevesloka commented on Makefile:
```diff
@@ -2,24 +2,66 @@
+build: build-dirs bin/$(BIN) bin/linux_$(ARCH)/$(BIN)
+
+# local developement binary (auto detect developer OS)
+bin/$(BIN): main.go
```
Why `go install` each time? If I'm just building, I don't see the need to put into path.

secat replied:
This comes from [Stackoverflow - What does go build build](http://stackoverflow.com/questions/30612611/what-does-go-build-build):

> So basically you can use `go build` as a check that the packages can be built (along with their dependencies) while `go install` also (permanently) installs the results in the proper folders of your `$GOPATH`.

For developing, and iterating it is faster. However, I can revert that change.

stevesloka commented on README.md:
```diff
@@ -9,89 +10,119 @@ Allow for Registry credentials to be refreshed inside your Kubernetes cluster vi
+2. Build: `make build`
 3. Test: `make test`
-4. Run on your machine: ` go run ./main.go --kubecfg-file=<pathToKubecfgFile> --use-kubernetes-cluster-service=false
+4. Run on your machine: `./bin/registry-creds --kubecfg-file=<pathToKubecfgFile>` or `go run ./main.go --kubecfg-file=<pathToKubecfgFile>`
```
I don't like the idea of having to build it first, then run. I like the workflow to just `go run`, see the output, stop it, then go run again. Do you have thoughts as to why this is better? I ends up being two steps each time you want to build.

secat replied:
Ok, I will rework my PR and remove these changes. The `Makefile` should look more like the original.

stevesloka commented on main.go:
```diff
@@ -43,21 +45,30 @@ import (
 )
 
 const (
+	dockerPrivateRegistryPasswordKey = "DOCKER_PRIVATE_REGISTRY_PASSWORD"
```
Can we combine with the other const?

secat replied:
Good idea, I will make the change.

stevesloka commented on main.go:
```diff
@@ -289,7 +355,7 @@ func main() {
 
 	log.Print("Using AWS Account: ", awsAccountID)
-	log.Printf("Using AWS Region: %s", *argAWSRegion)
+	log.Print("Using AWS Region: ", *argAWSRegion)
```
Why take out the printf?

secat replied:
In order to be consistent with other `Print`, I can revert this change.

stevesloka replied:
Oh I see your point, you can keep it, makes sense!

secat commented:
@stevesloka I addressed all previous comments.

I reverted almost all my changes in the Makefile. I am working on a windows platform and I did not realize that this project was favouring `go run` in order to run the binary outside of a docker container.

You should be able to build on your machine again. Sorry for that.

stevesloka commented:
Hey @secat I just merged! Thanks so much for the PR! This is really helpful!
